### PR TITLE
face_recognition_demo: add options to override the FD input size

### PR DESCRIPTION
**Pull request by maozhong1**
background: original face detection model have 300x300 input, it could not detect small faces in some use case(e.g, smart classroom).  we need add function to reshape fd-input-shape in demo app to detect more smaller faces.  This commit used to fulfill this requirement

### CODE REVIEW

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

File: demos/python_demos/face_recognition_demo/face_detector.py [hunk 1]
```diff
@@ -46,14 +46,19 @@ def clip(self, width, height):
...
         assert len(model.outputs) == 1, "Expected 1 output blob"
         self.input_blob = next(iter(model.inputs))
         self.output_blob = next(iter(model.outputs))
-        self.input_shape = model.inputs[self.input_blob].shape
+        #add reshape function to detect more small faces in image
+        if (input_h and input_w ):
+            self.input_shape = [1, 3, input_h, input_w]
+        else:
+            self.input_shape = model.inputs[self.input_blob].shape
+        print("face detector input shape:", self.input_shape)
```
Comment (Wovchena) [inline demos/python_demos/face_recognition_demo/face_detector.py]: * As for me, this changes are not required, because `model.inputs[self.input_blob].shape` changes when reshape is called.

* The demo uses `log` instead of `print`. Is it useful to print the values that a user specified? They just passed to print unchanged (well, concatenated with [1, 3]).

File: demos/python_demos/face_recognition_demo/face_recognition_demo.py [hunk 2]
```diff
@@ -70,7 +70,14 @@ def build_argparser():
                         help="Path to the Facial Landmarks Regression model XML file")
     models.add_argument('-m_reid', metavar="PATH", default="", required=True,
                         help="Path to the Face Reidentification model XML file")
-
```
Comment (Wovchena) [inline demos/python_demos/face_recognition_demo/face_recognition_demo.py]: I suggest returning that blank line back before `infer = parser.add_argument_group('Inference options')`.

File: demos/python_demos/face_recognition_demo/face_recognition_demo.py [hunk 3]
```diff
@@ -70,7 +70,14 @@ def build_argparser():
...
+    models.add_argument('-fd_iw', '--fd_input_width', default=0, type=int,
+                         help="(optional) specified the input shape of detection model " \
+                         "(default: use default input shape of model). Both -fd_iw and -fd_ih parameters " \
+                         "should be specified for reshape.")
+    models.add_argument('-fd_ih', '--fd_input_height', default=0, type=int,
+                         help="(optional) specified the input shape of detection model " \
```
Comment (Wovchena) [inline demos/python_demos/face_recognition_demo/face_recognition_demo.py]: * Following demo's style, it should be `help="(optional) Specify the input shape of detection model " \` or just `help="(optional) Height of detection model's input " \`. The same goes for `--fd_input_width`.

* Please, update `README.md` accordingly.

File: demos/python_demos/face_recognition_demo/face_recognition_demo.py [hunk 4]
```diff
@@ -122,12 +129,16 @@ def __init__(self, args):
 
         log.info("Loading models")
         face_detector_net = self.load_model(args.m_fd)
+        if args.fd_input_height and args.fd_input_width :
```
Comment (Wovchena) [inline demos/python_demos/face_recognition_demo/face_recognition_demo.py]: A user should get a warning if only `fd_input_height` or `fd_input_width` was provided.

File: demos/python_demos/face_recognition_demo/requirements.txt [hunk 5]
```diff
@@ -1,3 +1,4 @@
 opencv-python>=3.4.0
 numpy>=1.11.0
 scipy>=1.1.0
+Pillow>=6.0.0
```
Comment (Wovchena) [inline demos/python_demos/face_recognition_demo/requirements.txt]: Why was this added? I did not notice that the demo used something from that package.

File: demos/python_demos/face_recognition_demo/face_detector.py (hunk 1 above)
Comment (maozhong1) [reply demos/python_demos/face_recognition_demo/face_detector.py]: yes, this change not need after reshape is called.  I will revert this change. thanks

File: demos/python_demos/face_recognition_demo/face_recognition_demo.py (hunk 3 above)
Comment (maozhong1) [reply demos/python_demos/face_recognition_demo/face_recognition_demo.py]: ok. thanks

**Comment by maozhong1** [comment]
Thanks Wovchena's good suggestions. I committed all my changes per your comments.

**Comment by Wovchena** [comment]
Jenkins please retry a build

File: demos/python_demos/face_recognition_demo/face_detector.py [hunk 6]
```diff
@@ -46,7 +46,7 @@ def clip(self, width, height):
             self.position[:] = clip(self.position, min, max)
             self.size[:] = clip(self.size, min, max)
 
-    def __init__(self, model, confidence_threshold=0.5, roi_scale_factor=1.15):
+    def __init__(self, model, confidence_threshold=0.5, roi_scale_factor=1.15,input_h=0, input_w=0):
```
Comment (IRDonch) [inline demos/python_demos/face_recognition_demo/face_detector.py]: These new parameters aren't used anywhere.

**Comment by maozhong1** [comment]
> You have made `face_detector.py` executable. Please revert that; it is not a script, so it should not be executable.

sorry for the mistake. I will correct it.
